src/test/java/de/komoot/photon/ReverseSearchRequestHandlerTest.java

```diff
@@ -61,11 +52,21 @@ public void testHandle() throws BadRequestException {
         ReverseRequestFactory mockReverseRequestFactory = Mockito.mock(ReverseRequestFactory.class);
         Request mockWebRequest = Mockito.mock(Request.class);
         ReflectionTestUtil.setFieldValue(reverseSearchRequestHandlerUnderTest, ReverseSearchRequestHandler.class, "reverseRequestFactory", mockReverseRequestFactory);
-        ReverseRequestHandlerFactory mockReverseRequestHandlerFactory = Mockito.mock(ReverseRequestHandlerFactory.class);
+
+        SimpleReverseRequestHandler mockSimpleReverseRequestHandler = new SimpleReverseRequestHandler(null) {
```

Comment:
This does not look prettier but powermockito is really hackish and makes migrating to JDK9 problematic for now as it seems. Furthermore we should create a better test infrastructure where less or zero mocking is necessary.